src/main.rs

```diff
@@ -69,6 +69,14 @@ impl Layout for LuaLayout {
             .get::<_, LuaFunction>("handle_layout")?
             .call::<_, LuaTable>(args)?;
 
+        let layout_name = self
+            .lua
+            .globals()
+            .get::<_, LuaFunction>("handle_metadata")?
+            .call::<_, String>(output)?;
```

Comment:
Since this function is called `handle_metadata`, I think it should return a table of properties. For now the only interesting property would be `layout_name`, but in the future more can be added (layout namespace?).

Reply:
I agree. Can you help me out with the syntax, how do I get the layout_name from the returned table?

```rust
        let metadata = self
            .lua
            .globals()
            .get::<_, LuaFunction>("handle_metadata")?
            .call::<_, LuaTable>(args)?;

        // No field `layout_name` on type `mlua::Table<' >` unknown field
        generated_layout.layout_name = metadata.layout_name;
```
The last line here does not compile

Reply:
The same way we get `handle_metadata` from the globals table, by using `.get()`.